local/node.go

```diff
@@ -38,27 +38,50 @@ type getConnFunc func(context.Context, node.Node) (net.Conn, error)
 // AvalancheGo binaries in tests
 type NodeProcess interface {
 	// Start this process
-	Start() error
+	Start(string, chan string) error
 	// Send a SIGTERM to this process
 	Stop() error
 	// Returns when the process finishes exiting
 	Wait() error
+	// Returns if the process is executing
+	Alive() bool
 }
 
 type nodeProcessImpl struct {
-	cmd *exec.Cmd
+	cmd              *exec.Cmd
+	running          bool
+	waitReturnCh     chan error
+	unexpectedStopCh chan string
 }
 
-func (p *nodeProcessImpl) Start() error {
-	return p.cmd.Start()
+func (p *nodeProcessImpl) Start(name string, unexpectedStopCh chan string) error {
+	p.unexpectedStopCh = unexpectedStopCh
+	startErr := p.cmd.Start()
+	p.running = true
+	p.waitReturnCh = make(chan error, 1)
+	go func() {
+		p.waitReturnCh <- p.cmd.Wait()
+		if p.running {
+			// Stop() was not called
+			p.running = false
+			p.unexpectedStopCh <- name
+		}
+	}()
+	return startErr
 }
 
 func (p *nodeProcessImpl) Wait() error {
-	return p.cmd.Wait()
+	return <-p.waitReturnCh
```

Comment:
If we make the changes I suggest for `Start()` (namely passing in an `onStopChan`) we don't really need this method. To wait for the node to stop, you'd just wait on that channel.

If we do keep this, it should return an error if `Start()` hasn't been called. I also think it might make sense for it to return `nil` immediately  if `Stop()` has already been called.

Right now, if you call wait twice, the second call will block forever. If we keep this method, we should close `p.waitReturnCh` in the goroutine in `Start`

Reply:
changed implementation in different direction but addressing this issues. I think node Wait() is needed because onUnexpectedStopChan is used by ANR local lib user, not by ANR local lib, which indeed uses Wait

Reply:
Probably I shall explain that the channel is meant to be provided to ANR local lib user (the server, or another app). The same channel is passed to all node processes and also passed to ANR local lib user,  so the name is used to identify which node is passing the message. ANR per se does not use the channel.